test/integration/statefulset/statefulset_test.go

```diff
@@ -349,3 +362,111 @@ func setPodsReadyCondition(t *testing.T, clientSet clientset.Interface, pods *v1
 		t.Fatalf("failed to mark all StatefulSet pods to ready: %v", err)
 	}
 }
+
+// add for issue: https://github.com/kubernetes/kubernetes/issues/108837
+func TestStatefulSetStatusWithPodFail(t *testing.T) {
+	// Set up an API server
+	h := &framework.APIServerHolder{Initialized: make(chan struct{})}
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		<-h.Initialized
```

Comment:
Why do you need this handler?

An alternative approach might be to just fail pod creations here. You could count how many of the pod creation requests are successful, and return failures after a threshold is reached.

Then you don't need all this quota controllers setup.

Reply:
> An alternative approach might be to just fail pod creations here. You could count how many of the pod creation requests are successful, and return failures after a threshold is reached.

Thanks for your advice. I added an admission control to fail the pod creation.